variant-inc/terragrunt-variant-apps
# Bumping TF provider to v4

# Description

Bumping AWS provider version to latest stable one v4.22.0 for all modules except s3 as there were some breaking changes introduced on v4.0.0 and it has to be tested further.
I've read through changelog and didn't catch anything breaking for other resources, also the test worked fine.
Had to update required version in submodules so the references/tags will be updated here once those PRs are merged, that's also why pre-commit fails so you can ignore that.

Fixes [1763](https://drivevariant.atlassian.net/browse/CLOUD-1763)

## Type of change

Please delete options that are not relevant.


## How Has This Been Tested?

1. initial deployment with provider version below v4.0.0
- [octopus](https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-api-luka/deployments/releases/0.1.0-demo-luka-2-0001.350/deployments/Deployments-42021?activeTab=taskSummary)
- check that infrastructure is deployed as defined ✅ 

2. deployment after upgrade to v4.22.0
- use actions-octopus with TVA version `1.0.1-f-cloud-1763-v4-0001.466` [commit](https://github.com/variant-inc/demo-python-flask-variant-api/commit/69d9beb005720966e667151819c95fd9663805ab#diff-96340d949a8e9d0db7a285463b8dc21622ccfdbbb9b7e92a604557c64d68d867R46-R49)
- [octopus](https://octopus.apps.ops-drivevariant.com/app#/Spaces-2/projects/demo-api-luka/deployments/releases/0.1.0-demo-luka-2-0001.351/deployments/Deployments-42003?activeTab=taskSummary)
- check that infrastructure is deployed as defined ✅ 


## Checklist
